server/ce2e/license_test.go

```diff
@@ -0,0 +1,34 @@
+package ce2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/mattermost/mattermost-plugin-msteams-sync/server/testutils/containere2e"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRequiresLicense(t *testing.T) {
+	t.Parallel()
+	mattermost, _, tearDown := containere2e.NewE2ETestPlugin(t, containere2e.WithoutLicense())
+	defer tearDown()
+
+	client, err := mattermost.GetAdminClient(context.Background())
+	require.NoError(t, err)
+
+	t.Run("without license", func(t *testing.T) {
+		plugins, _, err := client.GetPlugins(context.Background())
+		require.NoError(t, err)
+
+		for _, plugin := range plugins.Active {
+			require.NotEqual(t, "com.mattermost.msteams-sync", plugin.Manifest.Id)
+		}
+
+		for _, plugin := range plugins.Inactive {
+			if plugin.Manifest.Id == "com.mattermost.msteams-sync" {
+				return
+			}
+		}
+		require.Fail(t, "failed to find plugin status at all")
```

Comment:
I'm not familiar with this test containers approach but seems that we are checking here that the plugin is inactive as it failed to start without the license. This works but wondering if there is another way, maybe at the plugin unit test level to check that it failed for the actual error of the license and not something else.